### DISCUSSION TITLE
Formatting dates with i18n

### DISCUSSION THREAD

**Discussion by artsiom-voitas** [spacedriveapp/spacedrive#2404] [Feature Request]
Currently, Spacedrive formats dates, for example on (spacedrive\interface\app\$libraryId\Explorer\Inspector\index.tsx):
`<MetaData icon={Eraser} label={t('modified')} value={formatDate(dateModified)} />`
using JavaScript capabilities, which makes it impossible to have date mapping in different languages. It is worth considering using the i18n functions provided in the [documentation](https://www.i18next.com/translation-function/formatting#datetime)

**Comment by utkubakir** [comment]
agreed, can go with a setting option too for configuring the format.

**Reply by artsiom-voitas** [reply]
Should I create an issue from this discussion?

**Reply by utkubakir** [reply]
sure

**Comment by AnimiVulpis** [comment]
Just to chime in here.
JavaScript has

> **Baseline** Widely available

localization functions. To be exact

- [Date.prototype.toLocaleDateString()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toLocaleDateString)
- [Date.prototype.toLocaleTimeString()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toLocaleTimeString)
- [Date.prototype.toLocaleString()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toLocaleString)

Those functions should offer everything you need without the need to pull in another library
